### PR TITLE
Adding macos and windows to CI/CD

### PR DESCRIPTION
**Pull request by sjrl**
## Description

Adding MacOS and Windows to the CI/CD workflow in addition to ubuntu

### CODE REVIEW

File: docs/py-modindex.html [hunk 1]
```diff
@@ -0,0 +1,286 @@
+<!DOCTYPE html>
```
Comment (jasonkhadka) [inline docs/py-modindex.html:1]: is this html file needed?

**Review by jasonkhadka — CHANGES_REQUESTED** [review]
there are 2 failed tests and not sure if the html file is needed in pr.

**Comment by sjrl** [comment]
@wuhu @jasonkhadka @blythed I also added tests for the Windows OS to see what would happen. It looks like we fail the Windows + 3.9. I wanted to ask if we ever plan on supporting Windows? If not I can remove them.

File: docs/py-modindex.html (hunk 1 above)
Comment (sjrl) [reply docs/py-modindex.html:1]: Yes this file was generated when building the docs and was missed in the last PR.

**Comment by jasonkhadka** [comment]
i think we should defenitely support windows. 
Also @sjrl  
You can run the changes to this CI/CD if you add the name of this branch in actions.yml
on:
  push:
    branches:
      - main

**Comment by sjrl** [comment]
> i think we should definitely support windows. Also @sjrl You can run the changes to this CI/CD if you add the name of this branch in actions.yml
> 
> ```
> on:
>   push:
>     branches:
>       - main
> ```

I think that is already present in `actions.yml` right? Or am I missing something?

**Comment by jasonkhadka** [comment]
> > i think we should definitely support windows. Also @sjrl You can run the changes to this CI/CD if you add the name of this branch in actions.yml
> > ```
> > on:
> >   push:
> >     branches:
> >       - main
> > ```
> 
> I think that is already present in `actions.yml` right? Or am I missing something?

yes, it is alright, sorry wrong suggestion here.

**Comment by sjrl** [comment]
We decided to leave this open until the Windows + 3.9 CI build is fixed. @wuhu @eramdiaz

**Comment by wuhu** [comment]
fixed in #330 and merged into there, closing here

**Pull request closed without merging**